src/format.rs

```diff
@@ -170,48 +170,122 @@ impl fmt::Display for HumanCount {
+    (0, ""),                  // no suffix
+];
+
 impl fmt::Display for HumanFloatCount {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         use fmt::Write;
-
-        // Use formatter's precision if provided, otherwise default to 4
-        let precision = f.precision().unwrap_or(4);
-        let num = format!("{:.*}", precision, self.0);
-
-        let (int_part, frac_part) = match num.split_once('.') {
-            Some((int_str, fract_str)) => (int_str.to_string(), fract_str),
-            None => (self.0.trunc().to_string(), ""),
-        };
-        let len = int_part.len();
-        for (idx, c) in int_part.chars().enumerate() {
-            let pos = len - idx - 1;
-            f.write_char(c)?;
-            if pos > 0 && pos % 3 == 0 {
-                f.write_char(',')?;
+        if f.alternate() {
+            // Find appropriate suffix
+            let (divisor, suffix) = HUMAN_FLOAT_COUNT_THRESHOLDS
+                .iter()
+                .find(|(threshold, _)| self.0.abs() >= *threshold)
+                .unwrap();
+
+            // Scale the number appropriately
+            let scaled_value = if *divisor > 0.0 {
+                self.0 / divisor
+            } else {
+                self.0
+            };
+
+            // If precision is manually set, use it
+            // Otherwise, calculate precision to show exactly 3 significant digits
+            let precision = f.precision().unwrap_or_else(|| {
+                if scaled_value.abs() < 10.0 {
+                    2
+                } else if scaled_value.abs() < 100.0 {
+                    1
+                } else {
+                    0
+                }
+            });
+
+            write!(f, "{:.*}{}", precision, scaled_value, suffix)?;
+        } else {
+            // Use formatter's precision if provided, otherwise default to 4
+            let precision = f.precision().unwrap_or(4);
+            let num = format!("{:.*}", precision, self.0);
+
+            let (int_part, frac_part) = match num.split_once('.') {
+                Some((int_str, fract_str)) => (int_str.to_string(), fract_str),
+                None => (self.0.trunc().to_string(), ""),
+            };
+            let len = int_part.len();
+            for (idx, c) in int_part.chars().enumerate() {
+                let pos = len - idx - 1;
+                f.write_char(c)?;
+                if pos > 0 && pos % 3 == 0 {
+                    f.write_char(',')?;
+                }
+            }
+            let frac_trimmed = frac_part.trim_end_matches('0');
+            if !frac_trimmed.is_empty() {
+                f.write_char('.')?;
+                f.write_str(frac_trimmed)?;
             }
-        }
-        let frac_trimmed = frac_part.trim_end_matches('0');
-        if !frac_trimmed.is_empty() {
-            f.write_char('.')?;
-            f.write_str(frac_trimmed)?;
         }
         Ok(())
     }
 }
 
+const HUMAN_FLOAT_COUNT_THRESHOLDS: [(f64, &str); 5] = [
```

Comment:
I don't think we need both this and `HUMAN_COUNT_TRESHOLDS`, presumably we can drive the float value from the integer one?

Reply:
The types are different, so they have to be defined separately unless I want to cast them every time the formatter runs. I could do something like this:

```rust
macro_rules! define_thresholds {
    ($(($num:expr, $suffix:expr)),*) => {
        const HUMAN_COUNT_THRESHOLDS: [(u64, &str); _] = [
            $(($num, $suffix),)*
        ];

        const HUMAN_FLOAT_COUNT_THRESHOLDS: [(f64, &str); _] = [
            $(($num as f64, $suffix),)*
        ];
    };
}

define_thresholds!(
    (1_000_000_000_000, "T"), // trillion
    (1_000_000_000, "B"),     // billion
    (1_000_000, "M"),         // million
    (1_000, "k"),             // thousand
    (0, "")                   // no suffix
);
```

but that feels a lot less maintainable than just having two arrays.